tests/blackbox_test.go

```diff
@@ -311,3 +338,32 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	}
 	return nil
 }
+
+// Identify and replace $uuid<num> with correct UUID
+// Variables with matching <num> should have identical UUIDs
+func replaceUUIDVars(t *testing.T, str string, UUIDmap map[int]string) string {
+	finalStr := str
+
+	pattern := regexp.MustCompile("\\$uuid([0-9]+)")
+	for _, match := range pattern.FindAllStringSubmatch(str, -1) {
+		if len(match) != 2 {
+			t.Fatalf("find all string submatch error: want length of 2, got length of %d", len(match))
```

Comment:
Rather than doing `Fatalf`'s inside this function return the error and handle it inside of the `outer` function.